core/src/main/java/org/mapfish/print/output/AbstractJasperReportOutputFormat.java

```diff
@@ -194,17 +194,22 @@ public final Print getJasperPrint(
       }
     }
     if (template.getJdbcUrl() != null) {
-      Connection connection;
-      if (template.getJdbcUser() != null) {
-        connection =
-            DriverManager.getConnection(
-                template.getJdbcUrl(), template.getJdbcUser(), template.getJdbcPassword());
-      } else {
-        connection = DriverManager.getConnection(template.getJdbcUrl());
-      }
-
-      print = fillManager.fill(jasperTemplateBuild.getAbsolutePath(), values.asMap(), connection);
+      Connection connection = null;
+      try {
+        if (template.getJdbcUser() != null) {
+          connection =
+              DriverManager.getConnection(
+                  template.getJdbcUrl(), template.getJdbcUser(), template.getJdbcPassword());
+        } else {
+          connection = DriverManager.getConnection(template.getJdbcUrl());
+        }
 
+        print = fillManager.fill(jasperTemplateBuild.getAbsolutePath(), values.asMap(), connection);
+      } finally {
+        if (connection != null) {
```

Comment:
To be safe better  add: "&& !connection.isClosed()"

Reply:
Resolving conversation, as proposed change has been committed